thorlcr/graph/thgraph.cpp

```diff
@@ -81,13 +82,20 @@ class CThorGraphResult : implements IThorResult, implements IRowWriter, public C
 public:
     IMPLEMENT_IINTERFACE;
 
-    CThorGraphResult(CActivityBase &_activity, IThorRowInterfaces *_rowIf, bool _distributed, unsigned spillPriority) : activity(_activity), rowIf(_rowIf), distributed(_distributed)
+    CThorGraphResult(CActivityBase &_activity, IThorRowInterfaces *_rowIf, ThorGraphResultType _resultType, unsigned spillPriority) : activity(_activity), rowIf(_rowIf), resultType(_resultType)
     {
         init();
+        EmptyRowSemantics emptyRowSemantics;
+        if (thorgraphresult_grouped == resultType)
```

Comment:
See other comments on the enumeration.  Since this is a bitfield this test should be 

    (thorgraphresult_grouped & resultType)

otherwise setting thorgraphresult_distributed will cause this to fail.